src/runtime/opencl.cpp

```diff
@@ -130,6 +134,24 @@ WEAK const char *halide_opencl_get_device_type(void *user_context) {
     return device_type;
 }
 
+WEAK void halide_opencl_set_build_options(const char *n) {
+    if (n) {
+        strncpy(build_options, n, 255);
```

Comment:
I know we're already using strncpy() somewhere else, but we probably shouldn't, since it has a bad failure mode (it can leave the destination not-null-terminated if the src string is too long).

Reply:
I've added explicit check to make it null-terminated here and in other places in this file.